main.c

```diff
@@ -532,12 +712,21 @@ int main(int argc, char *argv[]) {
 	if (state.output_manager == NULL) {
 		fprintf(stderr, "compositor doesn't support "
 			"wlr-output-management-unstable-v1\n");
-		return EXIT_FAILURE;
+		goto error;
 	}
 
-	while (state.running && wl_display_dispatch(display) != -1) {
-		// This space intentionally left blank
+	int ret = loop(&state);
+	goto done;
```

Comment:
This `goto` spaghetti is a little confusing. Maybe remove the `error` label, and assign `ret = EXIT_FAILURE` at each error site?

Reply:
As is it looks pretty ok I think?

Reply:
The gotos are interleaved which makes the logic tricky to follow.

Reply:
This appears to be fixed now.